File: tests/fixtures/sample.php
<?php

namespace App;

use App\Utils\Logger;

class UserService {
    public function greet(string $user): void {
        $this->helper();
        Logger::log($user);
    }

    private function helper(): int {
        return strlen("ok");
    }
}

$svc = new UserService();
$svc->greet("alice");

File: tests/fixtures/sample.py
import functools
from dataclasses import dataclass


def traced(fn):
    @functools.wraps(fn)
    def wrapper(*args, **kwargs):
        return fn(*args, **kwargs)

    return wrapper


@dataclass
class UserService:
    name: str

    @traced
    def greet(self, user: str) -> str:
        return f"{self.name}:{user}"


def build_service(name: str) -> UserService:
    svc = UserService(name)
    return svc
